File: _experimental/_ocaml/unit_test/demo/test/test_calculator.ml
open OUnit2
module Calculator = Demo.Calculator

let test_calculator = "Calculator Tests" >:::
  [
    "test_add" >:: (fun _ ->
      assert_equal 5 (Calculator.add 2 3);
    );

    "test_subtract" >:: (fun _ ->
      assert_equal 1 (Calculator.subtract 3 2);
    );
  ]

let () =
  run_test_tt_main test_calculator

File: _experimental/_c/unit_test/demo/test_calculator.c
#include "includes/unity.h"
#include "calculator.h"

void setUp(void)
{
    // This function can be used to set up any preconditions for the tests
}

void tearDown(void)
{
    // This function can be used to clean up after tests
}
// Test case for the add() function
void test_add(void)
{
    int expected = 5;
    int actual = add(2, 3);
    TEST_ASSERT_EQUAL_INT(5, actual);
}

void test_subtract(void)
{
    int expected = 1;
    int actual = subtract(3, 2);
    TEST_ASSERT_EQUAL_INT(expected, actual);
}

int main(void)
{
    UnityBegin("test_calculator.c");

    RUN_TEST(test_add);
    RUN_TEST(test_subtract);

    return UnityEnd();
}